adsingestp/parsers/jats.py

```diff
@@ -190,24 +191,58 @@ def parse(self, article_metadata):
         # JATS puts author data in <contrib-group>, giving individual authors in each <contrib>
         for art_group in art_contrib_groups:
             art_contrib_group = art_group.extract()
-            contribs_raw = art_contrib_group.find_all("contrib")
+
+            # extract <contrib> from each <contrib-group>
+            contribs_raw_init = []
+            if art_contrib_group.find_all("contrib", recursive=False):
+                contribs_raw_init = art_contrib_group.find_all("contrib", recursive=False)
+
             default_key = "ALLAUTH"
+
+            # cycle through <contrib> to check if a <collab> is listed in the same level as an author an has multiple authors nested under it; targeted for Springer
+            contribs_raw = []
+            collabflag = None
+            for i in range(len(contribs_raw_init)):
+                if contribs_raw_init[i].find("collab"):
+                    # find nested collab authors and unnest them
+                    if contribs_raw_init[i].find("collab").find("institution"):
+                        # save the author position of collab to add collab as an author later
+                        collabflag = i
+                        collab_authors = contribs_raw_init[i].find_all("contrib")
+                        # add new collab tag to each unnested author
+                        for ca in collab_authors:
+                            collabtag = copy(
+                                contribs_raw_init[collabflag].find("collab").find("institution")
+                            )
+                            ca.append(collabtag)
+                            contribs_raw.append(ca)
+                    else:
+                        contribs_raw.append(contribs_raw_init[i])
+
+                else:
+                    contribs_raw.append(contribs_raw_init[i])
+
             num_contribs = len(contribs_raw)
+            count = 0
             for contrib in contribs_raw:
```

Comment:
Ok, another thought on how to structure this and fold in the for loop up above with this one - what if you make this loop over `art_contrib_group.find_all("contrib", recursive=False)` - that'll get you all the top-level authors. Then, within the loop (at the top), do a `contrib.find("collab")` to see if the collab is at that author position. If so, do a `contrib.find_all("contrib")` to get the nested authors. You can add them to the top-level list of contribs like this:
```
for nested_contrib in nested_contribs:
        contrib.insert_after(nested_contrib.extract())
```
You'll need to do an insert_after for the collab itself too. That *should* make the nested collab authors get dealt with on the next loop iteration, but you'll have to double check that

Reply:
I took your advice and attempted a modification. It even caught some collabs that I missed the first time around. It works well now.